pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1108,41 +1132,36 @@ int tracepoint__sched__sched_process_exec(struct bpf_raw_tracepoint_args *ctx)
     p.task_info->recompute_scope = true;
 
     struct linux_binprm *bprm = (struct linux_binprm *) ctx->args[2];
-    if (bprm == NULL) {
+    if (bprm == NULL)
         return -1;
-    }
-    struct file *file = get_file_ptr_from_bprm(bprm);
-    void *file_path = get_path_str(__builtin_preserve_access_index(&file->f_path));
 
-    proc_info_t *proc_info = bpf_map_lookup_elem(&proc_info_map, &p.event->context.task.host_pid);
-    if (proc_info == NULL) {
+    proc_info_t *pinfo = bpf_map_lookup_elem(&proc_info_map, &p.event->context.task.host_pid);
+    if (pinfo == NULL) {
         // entry should exist in proc_map (init_program_data should have set it otherwise)
         tracee_log(ctx, BPF_LOG_LVL_WARN, BPF_LOG_ID_MAP_LOOKUP_ELEM, 0);
         return 0;
     }
 
-    proc_info->new_proc = true;
+    pinfo->new_proc = true;
+
+    struct file *file = get_file_ptr_from_bprm(bprm);
+    void *file_path = get_path_str(__builtin_preserve_access_index(&file->f_path));
 
-    // extract the binary name to be used in should_trace
-    __builtin_memset(proc_info->binary.path, 0, MAX_BIN_PATH_SIZE);
-    bpf_probe_read_str(proc_info->binary.path, MAX_BIN_PATH_SIZE, file_path);
-    proc_info->binary.mnt_id = p.event->context.task.mnt_id;
+    // Extract the binary name to be used in should_trace
+    __builtin_memset(pinfo->binary.path, 0, MAX_BIN_PATH_SIZE);
+    bpf_probe_read_str(pinfo->binary.path, MAX_BIN_PATH_SIZE, file_path);
+    pinfo->binary.mnt_id = p.event->context.task.mnt_id;
 
     if (!should_trace(&p))
```

Comment:
So in the control plane programs you can see I don't do these checks, so that we always have the full image of the system even if it's not in scope (since as I understand it that should only apply to the events, but we may need the data of the whole system anyway for tracee internally).
Now, it is possible that I got it wrong and we should add it there, but overall, this is related to your choice of not separating the control plane program. I will get to it in another comment.

Reply:
Yep that's a good question. Wether we should construct the process tree using all information we are capable of having or just the ones being filtered.

Reply:
Personally I'd err on consistency for now. So that check should be moved near the `should_submit` check (unless we go for program separation).

Reply:
> I think we probably want all information, as the process doing security related things is indicating on its parent and higher ancestors, which we want to have info on as well

That was my thought too.